spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/DiscoveryClientOptionalArgsConfiguration.java

```diff
@@ -83,6 +86,15 @@ public static void setupTLS(AbstractDiscoveryClientOptionalArgs<?> args, TlsProp
 		}
 	}
 
+	@Bean
+	EurekaClientHttpRequestFactorySupplier.RequestConfigCustomizer httpComponentsProtocolUpgradeRequestCustomizer(
```

Comment:
This should have a `@ConditionalOnMissingBean` on it - we want the users to be able to override programmatically.